src/Icon/Icon.native.js

```diff
@@ -50,3 +48,12 @@ const styles = StyleSheet.create({
     },
   },
 });
+
+const iconSize = (size: Size) =>
+  StyleSheet.create({
```

Comment:
It would be nice to avoid recreating stylesheet with every render.

What about adding something like below to `styles`
```
small: {
    fontSize: parseFloat(defaultTokens.widthIconSmall),
},
medium: {
    fontSize: parseFloat(defaultTokens.widthIconMedium),
},
large: {
    fontSize: parseFloat(defaultTokens.widthIconLarge),
},
```
and then use it like
`<RNText style={[styles.icon, styles[size], { color }]}>`

I understand that this approach can seems to look more 'hardcoded', but I don't think we will be expanding size options anytime soon.